[2025.11.0.b1] ModuleNotFoundError: No module named 'certifi'

### The problem

Trying to compile a simple ESP32 configuration, compilation fails with `ModuleNotFoundError: No module named 'certifi'`

The same yaml fails to compile with 2025.11-dev and 2025.12-dev with the same error.

### Which version of ESPHome has the issue?

2025.11.0b1

### What type of installation are you using?

Home Assistant Add-on

### What platform are you using?

ESP32

### Component causing the issue

_No response_

### YAML Config

```yaml
esphome:
  name: esp32-test
  friendly_name: esp32-test

esp32:
  board: esp32dev
  framework:
    type: esp-idf
    # version: 5.5.1
    # platform_version: https://github.com/pioarduino/platform-espressif32.git#develop

# Enable logging
logger:

# Enable Home Assistant API
api:
  encryption:
    key: "acGSCkL4YDaB7lrhvL949JbvxbElk7Q+XjGsHX4vmNM="

ota:
  - platform: esphome
    password: !secret ota_password

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: !secret ap_ssid
    password: !secret ap_password

captive_portal:
```

### Anything in the logs that might be useful for us?

```txt
2025.11.0b1
NFO ESPHome 2025.11.0b1
INFO Reading configuration /config/esphome/esp32-test.yaml...
INFO Generating C++ source...
INFO Compiling app... Build path: /data/build/esp32-test
********************************************************************************
If you like PlatformIO, please:
- star it on GitHub > https://github.com/platformio/platformio-core
- follow us on LinkedIn to stay up-to-date on the latest project news > https://www.linkedin.com/company/platformio/
- try PlatformIO IDE for embedded development > https://platformio.org/platformio-ide
********************************************************************************

Processing esp32-test (board: esp32dev; framework: espidf; platform: https://github.com/pioarduino/platform-espressif32/releases/download/55.03.31-1/platform-espressif32.zip)
--------------------------------------------------------------------------------
Platform Manager: Installing https://github.com/pioarduino/platform-espressif32/releases/download/55.03.31-1/platform-espressif32.zip
INFO Installing https://github.com/pioarduino/platform-espressif32/releases/download/55.03.31-1/platform-espressif32.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Platform Manager: espressif32@55.3.31 has been installed!
INFO espressif32@55.3.31 has been installed!
INFO tool-esp_install not installed, installing version https://github.com/pioarduino/esp_install/releases/download/v5.3.2/esp_install-v5.3.2.zip
INFO Installing tool-esp_install version https://github.com/pioarduino/esp_install/releases/download/v5.3.2/esp_install-v5.3.2.zip
Tool Manager: Installing https://github.com/pioarduino/esp_install/releases/download/v5.3.2/esp_install-v5.3.2.zip
INFO Installing https://github.com/pioarduino/esp_install/releases/download/v5.3.2/esp_install-v5.3.2.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-esp_install@5.3.2 has been installed!
INFO tool-esp_install@5.3.2 has been installed!
INFO tool-esp_install successfully installed and verified
Created pioarduino Python virtual environment using uv: /root/.platformio/penv
Warning: No internet connection detected, Python dependency check will be skipped.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'certifi'
Error: Failed to obtain certifi path from the virtual environment: Command '['/root/.platformio/penv/bin/python', '-c', 'import certifi; print(certifi.where())']' returned non-zero exit status 1.
INFO Package configuration completed successfully
Tool Manager: Installing https://github.com/pioarduino/esp-idf/releases/download/v5.5.1/esp-idf-v5.5.1.tar.xz
INFO Installing https://github.com/pioarduino/esp-idf/releases/download/v5.5.1/esp-idf-v5.5.1.tar.xz
Downloading  [####################################]  100%          
Unpacking  [####################################]  100%          
Tool Manager: framework-espidf@3.50501.0 has been installed!
INFO framework-espidf@3.50501.0 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/xtensa-esp-elf-14.2.0_20241119.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/xtensa-esp-elf-14.2.0_20241119.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: toolchain-xtensa-esp-elf@14.2.0+20241119 has been installed!
INFO toolchain-xtensa-esp-elf@14.2.0+20241119 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/esptoolpy-v5.1.0.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/esptoolpy-v5.1.0.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-esptoolpy@5.1.0 has been installed!
INFO tool-esptoolpy@5.1.0 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/contrib-piohome-3.4.4.tar.gz
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/contrib-piohome-3.4.4.tar.gz
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: contrib-piohome@3.4.4 has been installed!
INFO contrib-piohome@3.4.4 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/mklittlefs-3.2.0-new.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/mklittlefs-3.2.0-new.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-mklittlefs@3.2.0 has been installed!
INFO tool-mklittlefs@3.2.0 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/cmake-4.0.3.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/cmake-4.0.3.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-cmake@4.0.3 has been installed!
INFO tool-cmake@4.0.3 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/esp-rom-elfs-20241011.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/esp-rom-elfs-20241011.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-esp-rom-elfs@2024.10.11 has been installed!
INFO tool-esp-rom-elfs@2024.10.11 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/ninja-1.13.1.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/ninja-1.13.1.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-ninja@1.13.1 has been installed!
INFO tool-ninja@1.13.1 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/scons-4.8.1.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/scons-4.8.1.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-scons@4.40801.0 has been installed!
INFO tool-scons@4.40801.0 has been installed!
Tool Manager: Installing platformio/tool-scons @ ~4.40801.0
INFO Installing platformio/tool-scons @ ~4.40801.0
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-scons@4.40801.0 has been installed!
INFO tool-scons@4.40801.0 has been installed!
Library Manager: Installing esphome/noise-c @ 0.1.10
INFO Installing esphome/noise-c @ 0.1.10
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Library Manager: noise-c@0.1.10 has been installed!
INFO noise-c@0.1.10 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
Library Manager: Installing esphome/libsodium @ 1.10020.7
INFO Installing esphome/libsodium @ 1.10020.7
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Library Manager: libsodium@1.10020.7 has been installed!
INFO libsodium@1.10020.7 has been installed!
INFO Installing tools via idf_tools.py (this may take several minutes)...
Tool Manager: Installing file:///root/.platformio/tools/tool-esptoolpy
INFO Installing file:///root/.platformio/tools/tool-esptoolpy
Tool Manager: tool-esptoolpy@5.1.0 has been installed!
INFO tool-esptoolpy@5.1.0 has been installed!
INFO Tool tool-esptoolpy successfully installed
Warning: No internet connection detected, Python dependency check will be skipped.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'certifi'
Error: Failed to obtain certifi path from the virtual environment: Command '['/root/.platformio/penv/bin/python', '-c', 'import certifi; print(certifi.where())']' returned non-zero exit status 1.
INFO Installing tools via idf_tools.py (this may take several minutes)...
ERROR idf_tools.py installation failed (rc=1). Tail:
Traceback (most recent call last):
  File "/data/cache/platformio/packages/tool-esp_install/tools/idf_tools.py", line 31, in <module>
    import certifi
ModuleNotFoundError: No module named 'certifi'
INFO Installing tools via idf_tools.py (this may take several minutes)...
ERROR idf_tools.py installation failed (rc=1). Tail:
Traceback (most recent call last):
  File "/data/cache/platformio/packages/tool-esp_install/tools/idf_tools.py", line 31, in <module>
    import certifi
ModuleNotFoundError: No module named 'certifi'
INFO Installing tools via idf_tools.py (this may take several minutes)...
ERROR idf_tools.py installation failed (rc=1). Tail:
Traceback (most recent call last):
  File "/data/cache/platformio/packages/tool-esp_install/tools/idf_tools.py", line 31, in <module>
    import certifi
ModuleNotFoundError: No module named 'certifi'
INFO Installing tools via idf_tools.py (this may take several minutes)...
ERROR idf_tools.py installation failed (rc=1). Tail:
Traceback (most recent call last):
  File "/data/cache/platformio/packages/tool-esp_install/tools/idf_tools.py", line 31, in <module>
    import certifi
ModuleNotFoundError: No module named 'certifi'
INFO Installing tools via idf_tools.py (this may take several minutes)...
ERROR idf_tools.py installation failed (rc=1). Tail:
Traceback (most recent call last):
  File "/data/cache/platformio/packages/tool-esp_install/tools/idf_tools.py", line 31, in <module>
    import certifi
ModuleNotFoundError: No module named 'certifi'
INFO Package configuration completed successfully
Warning: No internet connection detected, Python dependency check will be skipped.
Error in package configuration: FileNotFoundError: [Errno 2] No such file or directory: '/root/.platformio/penv/bin/uv'
TypeError: cannot unpack non-iterable NoneType object:
  File "/usr/local/lib/python3.12/site-packages/platformio/builder/main.py", line 173:
    env.SConscript("$BUILD_SCRIPT")
  File "/data/cache/platformio/packages/tool-scons/scons-local-4.8.1/SCons/Script/SConscript.py", line 620:
    return _SConscript(self.fs, *files, **subst_kw)
  File "/data/cache/platformio/packages/tool-scons/scons-local-4.8.1/SCons/Script/SConscript.py", line 280:
    exec(compile(scriptdata, scriptname, 'exec'), call_stack[-1].globals)
  File "/data/cache/platformio/platforms/espressif32/builder/main.py", line 49:
    PYTHON_EXE, esptool_binary_path = platform.setup_python_env(env)
========================= [FAILED] Took 38.46 seconds =========================
```

### Additional information

_No response_

platformio wrongly assumes No internet Connection (https://github.com/pioarduino/platform-espressif32/issues/328).

It has been fixed in 55.03.33, but using this version throws another error
```

INFO ESPHome 2025.11.0b1
INFO Reading configuration /config/esphome/esp32-test.yaml...
WARNING The selected platform version is not the recommended one. If there are connectivity or build issues please remove the manual version.
WARNING The selected platform version is not the recommended one. If there are connectivity or build issues please remove the manual version.
INFO Generating C++ source...
INFO Compiling app... Build path: /data/build/esp32-test
********************************************************************************
If you like PlatformIO, please:
- star it on GitHub > https://github.com/platformio/platformio-core
- follow us on LinkedIn to stay up-to-date on the latest project news > https://www.linkedin.com/company/platformio/
- try PlatformIO IDE for embedded development > https://platformio.org/platformio-ide
********************************************************************************

Processing esp32-test (board: esp32dev; framework: espidf; platform: https://github.com/pioarduino/platform-espressif32/releases/download/55.03.33/platform-espressif32.zip)
--------------------------------------------------------------------------------
Platform Manager: Installing https://github.com/pioarduino/platform-espressif32/releases/download/55.03.33/platform-espressif32.zip
INFO Installing https://github.com/pioarduino/platform-espressif32/releases/download/55.03.33/platform-espressif32.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Platform Manager: espressif32@55.3.33 has been installed!
INFO espressif32@55.3.33 has been installed!
INFO tool-esp_install not installed, installing version https://github.com/pioarduino/esp_install/releases/download/v5.3.2/esp_install-v5.3.2.zip
INFO Installing tool-esp_install version https://github.com/pioarduino/esp_install/releases/download/v5.3.2/esp_install-v5.3.2.zip
Tool Manager: Installing https://github.com/pioarduino/esp_install/releases/download/v5.3.2/esp_install-v5.3.2.zip
INFO Installing https://github.com/pioarduino/esp_install/releases/download/v5.3.2/esp_install-v5.3.2.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-esp_install@5.3.2 has been installed!
INFO tool-esp_install@5.3.2 has been installed!
INFO tool-esp_install successfully installed and verified
Created pioarduino Python virtual environment using uv: /root/.platformio/penv
INFO Package configuration completed successfully
Tool Manager: Installing https://github.com/pioarduino/esp-idf/releases/download/v5.5.1/esp-idf-v5.5.1.tar.xz
INFO Installing https://github.com/pioarduino/esp-idf/releases/download/v5.5.1/esp-idf-v5.5.1.tar.xz
Downloading  [####################################]  100%          
Unpacking  [####################################]  100%          
Tool Manager: framework-espidf@3.50501.0 has been installed!
INFO framework-espidf@3.50501.0 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/xtensa-esp-elf-14.2.0_20250730.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/xtensa-esp-elf-14.2.0_20250730.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: toolchain-xtensa-esp-elf@14.2.0+20250730 has been installed!
INFO toolchain-xtensa-esp-elf@14.2.0+20250730 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/esptoolpy-v5.1.0.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/esptoolpy-v5.1.0.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-esptoolpy@5.1.0 has been installed!
INFO tool-esptoolpy@5.1.0 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/contrib-piohome-3.4.4.tar.gz
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/contrib-piohome-3.4.4.tar.gz
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: contrib-piohome@3.4.4 has been installed!
INFO contrib-piohome@3.4.4 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/mklittlefs-3.2.0-new.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/mklittlefs-3.2.0-new.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-mklittlefs@3.2.0 has been installed!
INFO tool-mklittlefs@3.2.0 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/cmake-4.0.3.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/cmake-4.0.3.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-cmake@4.0.3 has been installed!
INFO tool-cmake@4.0.3 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/esp-rom-elfs-20241011.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/esp-rom-elfs-20241011.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-esp-rom-elfs@2024.10.11 has been installed!
INFO tool-esp-rom-elfs@2024.10.11 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/ninja-1.13.1.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/ninja-1.13.1.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-ninja@1.13.1 has been installed!
INFO tool-ninja@1.13.1 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/scons-4.8.1.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/scons-4.8.1.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-scons@4.40801.0 has been installed!
INFO tool-scons@4.40801.0 has been installed!
Tool Manager: Installing platformio/tool-scons @ ~4.40801.0
INFO Installing platformio/tool-scons @ ~4.40801.0
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-scons@4.40801.0 has been installed!
INFO tool-scons@4.40801.0 has been installed!
Library Manager: Installing esphome/noise-c @ 0.1.10
INFO Installing esphome/noise-c @ 0.1.10
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Library Manager: noise-c@0.1.10 has been installed!
INFO noise-c@0.1.10 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
Library Manager: Installing esphome/libsodium @ 1.10020.7
INFO Installing esphome/libsodium @ 1.10020.7
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Library Manager: libsodium@1.10020.7 has been installed!
INFO libsodium@1.10020.7 has been installed!
INFO Installing tools via idf_tools.py (this may take several minutes)...
Tool Manager: Installing file:///root/.platformio/tools/tool-esptoolpy
INFO Installing file:///root/.platformio/tools/tool-esptoolpy
Tool Manager: tool-esptoolpy@5.1.0 has been installed!
INFO tool-esptoolpy@5.1.0 has been installed!
INFO Tool tool-esptoolpy successfully installed
INFO Installing tools via idf_tools.py (this may take several minutes)...
Tool Manager: Installing file:///root/.platformio/tools/toolchain-xtensa-esp-elf
INFO Installing file:///root/.platformio/tools/toolchain-xtensa-esp-elf
Tool Manager: toolchain-xtensa-esp-elf@14.2.0+20250730 has been installed!
INFO toolchain-xtensa-esp-elf@14.2.0+20250730 has been installed!
INFO Tool toolchain-xtensa-esp-elf successfully installed
INFO Installing tools via idf_tools.py (this may take several minutes)...
Tool Manager: Installing file:///root/.platformio/tools/tool-mklittlefs
INFO Installing file:///root/.platformio/tools/tool-mklittlefs
Tool Manager: tool-mklittlefs@3.2.0 has been installed!
INFO tool-mklittlefs@3.2.0 has been installed!
INFO Tool tool-mklittlefs successfully installed
INFO Installing tools via idf_tools.py (this may take several minutes)...
Tool Manager: Installing file:///root/.platformio/tools/tool-cmake
INFO Installing file:///root/.platformio/tools/tool-cmake
Tool Manager: tool-cmake@4.0.3 has been installed!
INFO tool-cmake@4.0.3 has been installed!
INFO Tool tool-cmake successfully installed
INFO Installing tools via idf_tools.py (this may take several minutes)...
Tool Manager: Installing file:///root/.platformio/tools/tool-ninja
INFO Installing file:///root/.platformio/tools/tool-ninja
Tool Manager: tool-ninja@1.13.1 has been installed!
INFO tool-ninja@1.13.1 has been installed!
INFO Tool tool-ninja successfully installed
INFO Installing tools via idf_tools.py (this may take several minutes)...
Tool Manager: Installing file:///root/.platformio/tools/tool-esp-rom-elfs
INFO Installing file:///root/.platformio/tools/tool-esp-rom-elfs
Tool Manager: tool-esp-rom-elfs@2024.10.11 has been installed!
INFO tool-esp-rom-elfs@2024.10.11 has been installed!
INFO Tool tool-esp-rom-elfs successfully installed
INFO Package configuration completed successfully
Installed esptool from tl-install path: /data/cache/platformio/packages/tool-esptoolpy
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
 - contrib-piohome @ 3.4.4 
 - framework-espidf @ 3.50501.0 (5.5.1) 
 - tool-cmake @ 4.0.3 
 - tool-esp-rom-elfs @ 2024.10.11 
 - tool-esptoolpy @ 5.1.0 
 - tool-mklittlefs @ 3.2.0 
 - tool-ninja @ 1.13.1 
 - tool-scons @ 4.40801.0 (4.8.1) 
 - toolchain-xtensa-esp-elf @ 14.2.0+20250730
Creating a new virtual environment for IDF Python dependencies using uv
Using Python 3.12.10 environment at: /root/.platformio/penv/.espidf-5.5.1
Installing ESP-IDF's Python dependencies with uv
Using Python 3.12.10 environment at: /root/.platformio/penv/.espidf-5.5.1
Resolved 27 packages in 189ms
Installed 27 packages in 42ms
 + annotated-types==0.7.0
 + certifi==2025.11.12
 + cffi==2.0.0
 + charset-normalizer==3.4.4
 + click==8.3.0
 + colorama==0.4.6
 + cryptography==44.0.3
 + esp-idf-kconfig==2.5.0
 + idf-component-manager==2.4.2
 + idna==3.11
 + jsonref==1.1.0
 + pycparser==2.23
 + pydantic==2.12.4
 + pydantic-core==2.41.5
 + pydantic-settings==2.12.0
 + pyparsing==3.2.5
 + python-dotenv==1.2.1
 + requests==2.32.5
 + requests-file==3.0.1
 + requests-toolbelt==1.0.0
 + ruamel-yaml==0.18.16
 + ruamel-yaml-clib==0.2.14
 + tqdm==4.67.1
 + truststore==0.10.4
 + typing-extensions==4.15.0
 + typing-inspection==0.4.2
 + urllib3==1.26.20
Reading CMake configuration...
-- Found Git: /usr/bin/git (found version "2.39.5")
-- git rev-parse returned 'fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).'
-- Minimal build - OFF
-- The C compiler identification is GNU 14.2.0
-- The CXX compiler identification is GNU 14.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /data/cache/platformio/packages/toolchain-xtensa-esp-elf/bin/xtensa-esp32-elf-gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /data/cache/platformio/packages/toolchain-xtensa-esp-elf/bin/xtensa-esp32-elf-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /data/cache/platformio/packages/toolchain-xtensa-esp-elf/bin/xtensa-esp32-elf-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building ESP-IDF components for target esp32
NOTICE: Dependencies lock doesn't exist, solving dependencies.
...NOTICE: Updating lock file at /data/build/esp32-test/dependencies.lock
NOTICE: Processing 3 dependencies:
NOTICE: [1/3] espressif/mdns (1.8.2)
NOTICE: [2/3] zorxx/multipart-parser (1.0.1)
NOTICE: [3/3] idf (5.5.1)
-- Project sdkconfig file /data/build/esp32-test/sdkconfig.esp32-test
/data/build/esp32-test/sdkconfig.esp32-test:2 CONFIG_COMPILER_OPTIMIZATION_DEFAULT was replaced with CONFIG_COMPILER_OPTIMIZATION_DEBUG 
/data/build/esp32-test/sdkconfig.esp32-test:9 CONFIG_ESP_TASK_WDT was replaced with CONFIG_ESP_TASK_WDT_INIT 
-- Compiler supported targets: xtensa-esp-elf
-- Configuring incomplete, errors occurred!

fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
warning: user value 0 on the int symbol LWIP_BRIDGEIF_MAX_PORTS (defined at /data/cache/platformio/packages/framework-espidf/components/lwip/Kconfig:1222) ignored due to being outside the active range ([1, 63]) -- falling back on defaults
CMake Error at /data/cache/platformio/packages/framework-espidf/tools/cmake/tool_version_check.cmake:36 (message):


  Tool doesn't match supported version from list ['esp-14.2.0_20241119']:
  /data/cache/platformio/packages/toolchain-xtensa-esp-elf/bin/xtensa-esp32-elf-gcc

  Please try to run 'idf.py fullclean' to solve it.

Call Stack (most recent call first):
  /data/cache/platformio/packages/framework-espidf/components/esp_common/project_include.cmake:10 (check_expected_tool_version)
  /data/cache/platformio/packages/framework-espidf/tools/cmake/build.cmake:467 (include)
  /data/cache/platformio/packages/framework-espidf/tools/cmake/build.cmake:718 (__build_process_project_includes)
  /data/cache/platformio/packages/framework-espidf/tools/cmake/project.cmake:740 (idf_build_process)
  CMakeLists.txt:3 (project)



========================= [FAILED] Took 90.71 seconds =========================
```

Revert to defaults, clean build files, the build succeeds.

--- a/esphome/components/esp32/__init__.py
+++ b/esphome/components/esp32/__init__.py
@@ -334,12 +334,14 @@ def _is_framework_url(source: str) -> str:
 #  - https://github.com/espressif/arduino-esp32/releases
 ARDUINO_FRAMEWORK_VERSION_LOOKUP = {
     "recommended": cv.Version(3, 3, 2),
-    "latest": cv.Version(3, 3, 2),
-    "dev": cv.Version(3, 3, 2),
+    "latest": cv.Version(3, 3, 4),
+    "dev": cv.Version(3, 3, 4),
 }
 ARDUINO_PLATFORM_VERSION_LOOKUP = {
-    cv.Version(3, 3, 2): cv.Version(55, 3, 31, "1"),
-    cv.Version(3, 3, 1): cv.Version(55, 3, 31, "1"),
+    cv.Version(3, 3, 4): cv.Version(55, 3, 31, "2"),
+    cv.Version(3, 3, 3): cv.Version(55, 3, 31, "2"),
+    cv.Version(3, 3, 2): cv.Version(55, 3, 31, "2"),
+    cv.Version(3, 3, 1): cv.Version(55, 3, 31, "2"),
     cv.Version(3, 3, 0): cv.Version(55, 3, 30, "2"),
     cv.Version(3, 2, 1): cv.Version(54, 3, 21, "2"),
     cv.Version(3, 2, 0): cv.Version(54, 3, 20),
@@ -357,8 +359,8 @@ ESP_IDF_FRAMEWORK_VERSION_LOOKUP = {
     "dev": cv.Version(5, 5, 1),
 }
 ESP_IDF_PLATFORM_VERSION_LOOKUP = {
-    cv.Version(5, 5, 1): cv.Version(55, 3, 31, "1"),
-    cv.Version(5, 5, 0): cv.Version(55, 3, 31, "1"),
+    cv.Version(5, 5, 1): cv.Version(55, 3, 31, "2"),
+    cv.Version(5, 5, 0): cv.Version(55, 3, 31, "2"),
     cv.Version(5, 4, 3): cv.Version(55, 3, 32),
     cv.Version(5, 4, 2): cv.Version(54, 3, 21, "2"),
     cv.Version(5, 4, 1): cv.Version(54, 3, 21, "2"),
@@ -373,9 +375,9 @@ ESP_IDF_PLATFORM_VERSION_LOOKUP = {
 # The platform-espressif32 version
 #  - https://github.com/pioarduino/platform-espressif32/releases
 PLATFORM_VERSION_LOOKUP = {
-    "recommended": cv.Version(55, 3, 31, "1"),
-    "latest": cv.Version(55, 3, 31, "1"),
-    "dev": cv.Version(55, 3, 31, "1"),
+    "recommended": cv.Version(55, 3, 31, "2"),
+    "latest": cv.Version(55, 3, 31, "2"),
+    "dev": cv.Version(55, 3, 31, "2"),
 }
 
 

